sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -171,6 +171,27 @@ trait FileFormat {
   def supportFieldName(name: String): Boolean = true
 }
 
+object FileFormat {
+
+  val FILE_PATH = "file_path"
+
+  val FILE_NAME = "file_name"
+
+  val FILE_SIZE = "file_size"
+
+  val FILE_MODIFICATION_TIME = "file_modification_time"
+
+  val METADATA_NAME = "_metadata"
+
+  // supported metadata columns for hadoop fs relation
+  val FILE_METADATA_COLUMNS: AttributeReference = MetadataAttribute(METADATA_NAME,
```

Comment:
I think this should be a `def`, otherwise all the file source metadata columns share one `ExprId`. This can cause issues if you scan two file sources and join them.